File: JavaScript/Stack/2390. Removing Stars From a String.js
/* 2390. Removing Stars From a String
Difficulty: Easy
Submission Link: https://leetcode.com/problems/removing-stars-from-a-string/submissions/1375560717/ */

var removeStars = function(s) {
    const stack = [];
    for (const c of s) {
        if (c === '*' && stack.length) {
            stack.pop();
        }
        else {
            stack.push(c);
        }
    }
    return stack.join('');
};

File: Python/Stack/2390. Removing Stars From a String.py
""" 2390. Removing Stars From a String
Difficulty: Easy
Submission Link: https://leetcode.com/problems/removing-stars-from-a-string/submissions/1375560983/ """

class Solution(object):
    def removeStars(self, s):
        stack = []
        for c in s:
            if c == '*' and stack:
                stack.pop()
            else:
                stack.append(c)
        return ''.join(stack)